pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,32 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// FindMatchingUntoleratedTaint checks if the given tolerations tolerates
```

Comment:
If there are no other users, just remove `TolerationsTolerateTaintsWithFilter`.

Reply:
`TolerationsTolerateTaintsWithFilter` is used in the daemon controller

Reply:
`TolerationsTolerateTaintsWithFilter` is being used in the daemon controller

Reply:
Please make `TolerationsTolerateTaintsWithFilter` call this function and add a comment saying it's deprecated and to call this instead.

Reply:
updated